docs/architecture/modernize-desktop/windows-migration.md

```diff
@@ -0,0 +1,524 @@
+references the full-trust app. When that application is launched, it doesn't
+run inside an app container, but instead it runs as the user as it normally
+would.
+
+After deployment, package files are marked read-only and heavily locked down by
+the operating system. Windows prevents apps from launching if these files are
+tampered with.
+
+##### File system
+
+The OS supports different levels of file system operations for packaged desktop
+applications, depending on the folder location.
+
+When trying to access the user's *AppData* folder, the system creates a private
+per-user, per-app location behind the scenes. This creates the illusion that the
+packaged application is editing the real *AppData* when it's actually modifying a
+private copy. By redirecting writes this way, the system can track all file
+modifications made by the app. It can then clean all those files when
+uninstalling reducing system "rot" and providing a better application removal
+experience for the user.
+
+`##### Registry`
+
+App packages contain a registry.dat file, which serves as the logical equivalent
+of `HKLM\Software` in the real registry. At runtime, this virtual registry
+merges the contents of this hive into the native system hive to provide a
+singular view of both.
+
+All writes are kept during package upgrade and only deleted when the application
+is uninstalled.
+
+##### Uninstallation
+
+When the user uninstalls a package, all files and folders located
+under `C:\Program Files\WindowsApps\package_name` are removed, as well as any
+redirected writes to AppData or the registry that were captured during the
+process.
+
+For details about how a packaged application handles installation, file access,
+registry, and uninstallation, see
+<https://docs.microsoft.com/windows/msix/desktop/desktop-to-uwp-behind-the-scenes>.
+
+You can get a complete list of things to check on
+<https://docs.microsoft.com/windows/msix/desktop/desktop-to-uwp-prepare>.
+
+## How to add WinRT APIs to your desktop project
+
+In this section, you can find a walkthrough on how to integrate Toast
+Notifications in an existing WPF application. Although it's simple from the
+code perspective, it helps illustrate the whole process. Notifications are one
+of the many available WinRT APIs available that you can use in .NET app. In this
+case, the API requires a Package Identity. This process is more straightforward
+if the APIs don't require Package Identity.
+
+Let's take an existing WPF sample app that reads files and shows its contents on
+the screen. The goal is to display a Toast Notification when the
+application starts.
+
+![Screenshot of the sample application running](media\windows-migration\sample-application.png)
+
+First, you should check in the following link whether the Windows 10 API that
+you'll use requires a Package Identity:
+
+<https://docs.microsoft.com/windows/apps/desktop/modernize/desktop-to-uwp-supported-api>
```

Comment:
For these bare links, is it important that they show up in print? otherwise, I'd recommend adding the title too for SEO purposes and better readability